src/commands/economy/farm.js

```diff
@@ -0,0 +1,410 @@
+const User = require('../../schemas/user-schema.js');
+
+module.exports = {
+	data: new SlashCommandBuilder()
+		.setName('farm')
+		.setNameLocalizations({
+			'pt-BR': 'fazenda',
+			'es-ES': 'granja',
+		})
+		.setDescription('Farm commands')
+		.setDMPermission(false)
+		.addSubcommand((subcommand) =>
+			subcommand
+				.setName('view')
+				.setNameLocalizations({
+					'pt-BR': 'ver',
+					'es-ES': 'ver',
+				})
+				.setDescription('View your farm')
+				.setDescriptionLocalizations({
+					'pt-BR': 'Veja sua fazenda',
+					'es-ES': 'Ve tu granja',
+				})
+		)
+		.addSubcommand((subcommand) =>
+			subcommand
+				.setName('plant')
+				.setNameLocalizations({
+					'pt-BR': 'plantar',
+					'es-ES': 'plantar',
+				})
+				.setDescription('Plant a crop')
+				.setDescriptionLocalizations({
+					'pt-BR': 'Plante uma safra',
+					'es-ES': 'Planta un cultivo',
+				})
+				.addStringOption((option) =>
+					option
+						.setName('crop')
+						.setNameLocalizations({
+							'pt-BR': 'cultura',
+							'es-ES': 'cultivo',
+						})
+						.setDescription('Crop to plant')
+						.setDescriptionLocalizations({
+							'pt-BR': 'Cultura para plantar',
+							'es-ES': 'Cultivo para plantar',
+						})
+						.setAutocomplete(true)
+				)
+		)
+		.addSubcommand((subcommand) =>
+			subcommand
+				.setName('water')
+				.setNameLocalizations({
+					'pt-BR': 'regar',
+					'es-ES': 'regar',
+				})
+				.setDescription('Water a crop')
+				.setDescriptionLocalizations({
+					'pt-BR': 'Regue uma safra',
+					'es-ES': 'Riega un cultivo',
+				})
+		)
+		.addSubcommand((subcommand) =>
+			subcommand
+				.setName('harvest')
+				.setNameLocalizations({
+					'pt-BR': 'colher',
+					'es-ES': 'cosechar',
+				})
+				.setDescription('Harvest a crop')
+				.setDescriptionLocalizations({
+					'pt-BR': 'Colha uma safra',
+					'es-ES': 'Cosecha un cultivo',
+				})
+		)
+		.addSubcommand((subcommand) =>
+			subcommand
+				.setName('uproot')
+				.setNameLocalizations({
+					'pt-BR': 'arrancar',
+					'es-ES': 'arrancar',
+				})
+				.setDescription('Uproot a crop')
+				.setDescriptionLocalizations({
+					'pt-BR': 'Arranque uma safra',
+					'es-ES': 'Arranca un cultivo',
+				})
+				.addIntegerOption((option) =>
+					option
+						.setName('plot')
+						.setNameLocalizations({
+							'pt-BR': 'safra',
+							'es-ES': 'cultivo',
+						})
+						.setDescription('Plot to uproot')
+						.setDescriptionLocalizations({
+							'pt-BR': 'Safra para arrancar',
+							'es-ES': 'Cultivo para arrancar',
+						})
+				)
+		),
+	execute: async ({ interaction, instance, subcommand, member, args }) => {
+		await interaction.deferReply().catch(() => {});
+		try {
+			var type = interaction.options.getSubcommand();
+		} catch {
+			var type = subcommand;
+		}
+
+		const MAX_PLOTS = 6;
+		const WATER_COOLDOWN = 60 * 60 * 1000;
+
+		const player = await User.findByIdAndUpdate(member.id, {}, { select: 'plots inventory', upsert: true, new: true });
```

Comment:
```suggestion
		const player = await User.findById(member.id, 'plots inventory', { upsert: true });
```
This query can be simplified to use just findById

Reply:
as discussed with the contributor offline, the query suggested actually does not insert a new document if there is not one there already